### DISCUSSION TITLE
How can i manage response of navigator.credentials.get error?

### DISCUSSION THREAD

**Discussion by MimiaBena** [passwordless-lib/fido2-net-lib#492] [Q&A] [answered]
Hello,
I'm using `credentials.get` for get the cred stored on my browser. I have the list but i can't manage 2 case:

1- credentials are presented to the user in a modal dialog . If the user makes a gesture outside of the dialog, the dialog closes without resolving or rejecting the Promise and passe to the cache `DOMException: The operation either timed out or was not allowed.`
2-  the user hasn't any cred stored and he click Ok. the dialog closes without resolving or rejecting the Promise and passe to the cache with `DOMException: The operation either timed out or was not allowed.`

How i can solve this?

Any help please

`await navigator.credentials.get({
      publicKey: options,
    })
    .then((res) => {
      console.log(res)
    }).catch((err) => {
      console.log(err)
    })`

Thanx

**Comment by Sora2455** [comment]
There isn't really anything to solve here. The user didn't give you a credential, so you don't get a credential. Just add an if statement to your catch statement, looking for that particular error, and show the user an error message along the lines of "No credential selected".

**Comment by MimiaBena** [comment]
i have any inforrmation to make if statement, for 2 cases, i have the same error.

**Comment by Sora2455** [comment]
Then you'll have to use an error message that's either vague enough for both, or explicitly mentions both possible causes.

**Comment by Regenhardt** [comment]
@MimiaBena have you found a solution? How about something like "No valid credential received from device, please try again or choose another option.", conveying both either cancelling the dialogue or just not having any credentials and also providing a way forward for the user?

**Comment by abergs** [comment, marked as answer]
This behaviour is by design. We have strict privacy requirements on the API - it's designed to never leak information about the user - if they have a credential or not, unless the user confirms the auth.
